src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2EOSSolver.cpp

```diff
@@ -105,6 +98,14 @@ CO2EOSSolver::solve( string const & name,
     real64 const newVar = var + update;
     real64 const newRes = (*f)( temp, pres, newVar );
 
+    // check convergence based on the magnitude of the residual
+    if( LvArray::math::abs( newRes ) < tolerance )
+    {
+      var = newVar;
+      newtonHasConverged = true;
+      break;
```

Comment:
I would have put a `return var;` as it makes clearer that you return from the function but this is totally right to keep and I don't know what is the policy in GEOS.